README example triggers glam version mismatch — use re-exported `cadrum::DVec3`

## Problem

The README example begins with:

```rust
use cadrum::Solid;
use glam::DVec3;
```

which encourages new users to add `glam = "0.xx"` to their own `Cargo.toml`. Because `cadrum`'s Cargo.toml declares `glam = ">=0.29"` but is internally resolved against a specific minor (currently 0.32), users pick any 0.x version and two incompatible `glam` crates end up in the dependency graph:

```
error[E0308]: mismatched types
   expected `cadrum::DVec3`, found `glam::DVec3`
note: there are multiple different versions of crate `glam` in the dependency graph
```

(Cargo treats 0.x minor bumps as breaking, so 0.29 and 0.32 never unify.)

## Root cause

`cadrum` already re-exports the right type — see `src/lib.rs:37`:

```rust
pub use glam::DVec3;
```

The README example just doesn't use it, so users don't discover the workaround without digging.

## Proposal

1. Update the README example to:
   ```rust
   use cadrum::{DVec3, Solid};
   ```
2. Add a short note that users should **not** add `glam` to their own `Cargo.toml`.
3. (Optional, additive) Broaden the re-export to `pub use glam;` so users can reach `Vec3`, `Quat`, `Mat4` etc. via `cadrum::glam::*` without pinning their own glam version.

## Impact

Eliminates the most common first-use failure. Fully backwards-compatible (docs-only + additive re-exports).

--- a/README.md
+++ b/README.md
@@ -27,6 +27,8 @@ Add this to your `Cargo.toml`:
 cadrum = "^0.6"
 ```
 
+> **Note**: do **not** add `glam` as a separate dependency. cadrum's public API takes `glam` types (`DVec3`, `DQuat`, …), so a mismatched `glam` minor version leads to `expected cadrum::DVec3, found glam::DVec3` errors. Use `cadrum::{DVec3, DQuat, ...}` (re-exported at the crate root) or `cadrum::glam::*` instead.
+
 ## Build
 
 `cargo build` automatically downloads a prebuilt OCCT 8.0.0-rc5 binary for the targets below.
@@ -62,8 +64,7 @@ cargo run --example 01_primitives
 ```rust
 //! Primitive solids: box, cylinder, sphere, cone, torus — colored and exported as STEP + SVG.
 
-use cadrum::Solid;
-use glam::DVec3;
+use cadrum::{DVec3, Solid};
 
 fn main() {
     let example_name = std::path::Path::new(file!()).file_stem().unwrap().to_str().unwrap();
@@ -110,8 +111,7 @@ cargo run --example 02_write_read
 ```rust
 //! Read and write: chain STEP, BRep text, and BRep binary round-trips with progressive rotation.
 
-use cadrum::{Solid, Transform};
-use glam::DVec3;
+use cadrum::{Compound, DVec3, Solid};
 use std::f64::consts::FRAC_PI_8;
 
 fn main() -> Result<(), cadrum::Error> {
@@ -186,8 +186,7 @@ cargo run --example 03_transform
 ```rust
 //! Transform operations: translate, rotate, scale, and mirror applied to a cone.
 
-use cadrum::Solid;
-use glam::DVec3;
+use cadrum::{DVec3, Solid};
 use std::f64::consts::PI;
 
 fn main() {
@@ -245,8 +244,7 @@ cargo run --example 04_boolean
 ```rust
 //! Boolean operations: union, subtract, and intersect between a box and a cylinder.
 
-use cadrum::{Solid, Transform};
-use glam::DVec3;
+use cadrum::{Compound, DVec3, Solid};
 
 fn main() -> Result<(), cadrum::Error> {
     let example_name = std::path::Path::new(file!()).file_stem().unwrap().to_str().unwrap();
@@ -305,8 +303,7 @@ cargo run --example 05_extrude
 //! - **L-beam**: L-shaped polygon extruded along Z
 //! - **Heart**: BSpline heart-shaped profile extruded along Z
 
-use cadrum::{BSplineEnd, Edge, Error, Solid};
-use glam::DVec3;
+use cadrum::{BSplineEnd, DVec3, Edge, Error, Solid};
 
 /// Square polygon → box (simplest extrude).
 fn build_box() -> Result<Solid, Error> {
@@ -401,8 +398,7 @@ cargo run --example 06_loft
 //! - **Morph**: square polygon → circle (cross-section shape transition)
 //! - **Tilted**: three non-parallel circular sections → twisted loft
 
-use cadrum::{Edge, Error, Solid};
-use glam::DVec3;
+use cadrum::{DVec3, Edge, Error, Solid};
 
 /// Two circles → frustum (minimal loft example).
 fn build_frustum() -> Result<Solid, Error> {
@@ -495,8 +491,7 @@ cargo run --example 07_sweep
 //!   toward a parallel auxiliary spine. Arbitrary twist control — e.g. a
 //!   helical `aux_spine` on a straight `spine` produces a twisted ribbon.
 
-use cadrum::{Compound, Edge, Error, ProfileOrient, Solid, Transform};
-use glam::DVec3;
+use cadrum::{Compound, DVec3, Edge, Error, ProfileOrient, Solid, Wire};
 
 // ==================== Component 1: M2 ISO screw ====================
 
@@ -566,11 +561,12 @@ fn build_u_pipe() -> Result<Vec<Solid>, Error> {
 
 // ==================== Component 3: Auxiliary-spine twisted ribbon ====================
 
-// 直線 spine を `Auxiliary(&[helix])` で掃引すると、各点で profile の tracked 軸が
-// 対応するヘリックス点を向くように回転される。pitch=h のヘリックスは [0, h] の
-// あいだにちょうど 360° 一周するので、平たい長方形 profile は 1 回捻れた
-// リボンになる — `Fixed` や `Torsion` だと直線 spine では profile は全く
-// 回転しないので、ねじれが見えれば Auxiliary が効いている証拠。
+// Sweeping a straight spine with `Auxiliary(&[helix])` rotates the tracked
+// axis of the profile at each point to face the corresponding helix point.
+// A pitch=h helix makes exactly one 360° turn over [0, h], so a flat
+// rectangular profile becomes a ribbon twisted once. With `Fixed` or
+// `Torsion` the profile wouldn't rotate along a straight spine — visible
+// twist is therefore proof that Auxiliary is in effect.
 fn build_twisted_ribbon() -> Result<Vec<Solid>, Error> {
 	let h = 8.0;
 	let aux_r = 3.0;
@@ -578,7 +574,7 @@ fn build_twisted_ribbon() -> Result<Vec<Solid>, Error> {
 	let spine = Edge::line(DVec3::ZERO, DVec3::Z * h)?;
 	let aux = Edge::helix(aux_r, h, h, DVec3::Z, DVec3::X)?;
 
-	// 平たい長方形 (10:1 アスペクト) — 円や正方形ではねじれが見えない。
+	// Flat rectangle (10:1 aspect) — circles or squares wouldn't reveal any twist.
 	let profile = Edge::polygon(&[DVec3::new(-2.0, -0.2, 0.0), DVec3::new(2.0, -0.2, 0.0), DVec3::new(2.0, 0.2, 0.0), DVec3::new(-2.0, 0.2, 0.0)])?;
 
 	let ribbon = Solid::sweep(&profile, &[spine], ProfileOrient::Auxiliary(&[aux]))?;
@@ -618,8 +614,7 @@ cargo run --example 08_bspline
 ```
 
 ```rust
-use cadrum::Solid;
-use glam::{DQuat, DVec3};
+use cadrum::{DQuat, DVec3, Solid};
 use std::f64::consts::TAU;
 
 // 2 field-period stellarator-like torus.

--- a/README.md
+++ b/README.md
@@ -27,8 +27,6 @@ Add this to your `Cargo.toml`:
 cadrum = "^0.6"
 ```
 
-> **Note**: do **not** add `glam` as a separate dependency. cadrum's public API takes `glam` types (`DVec3`, `DQuat`, …), so a mismatched `glam` minor version leads to `expected cadrum::DVec3, found glam::DVec3` errors. Use `cadrum::{DVec3, DQuat, ...}` (re-exported at the crate root) or `cadrum::glam::*` instead.
-
 ## Build
 
 `cargo build` automatically downloads a prebuilt OCCT 8.0.0-rc5 binary for the targets below.
@@ -363,16 +361,13 @@ fn main() -> Result<(), Error> {
 
 	let result = [box_solid, oblique, l_beam, heart];
 
-	let step_path = format!("{example_name}.step");
-	let mut f = std::fs::File::create(&step_path).expect("failed to create STEP file");
+	let mut f = std::fs::File::create(format!("{example_name}.step")).expect("failed to create STEP file");
 	cadrum::write_step(&result, &mut f).expect("failed to write STEP");
-	println!("wrote {step_path}");
 
-	let svg_path = format!("{example_name}.svg");
-	let mut f = std::fs::File::create(&svg_path).expect("failed to create SVG file");
+	let mut f = std::fs::File::create(format!("{example_name}.svg")).expect("failed to create SVG file");
 	cadrum::mesh(&result, 0.5).and_then(|m| m.write_svg(DVec3::new(1.0, 1.0, 1.0), true, false, &mut f)).expect("failed to write SVG");
-	println!("wrote {svg_path}");
 
+	println!("wrote {example_name}.step / {example_name}.svg");
 	Ok(())
 }
 
@@ -441,16 +436,13 @@ fn main() -> Result<(), Error> {
 
 	let result = [frustum, morph, tilted];
 
-	let step_path = format!("{example_name}.step");
-	let mut f = std::fs::File::create(&step_path).expect("failed to create STEP file");
+	let mut f = std::fs::File::create(format!("{example_name}.step")).expect("failed to create STEP file");
 	cadrum::write_step(&result, &mut f).expect("failed to write STEP");
-	println!("wrote {step_path}");
 
-	let svg_path = format!("{example_name}.svg");
-	let mut f = std::fs::File::create(&svg_path).expect("failed to create SVG file");
+	let mut f = std::fs::File::create(format!("{example_name}.svg")).expect("failed to create SVG file");
 	cadrum::mesh(&result, 0.5).and_then(|m| m.write_svg(DVec3::new(1.0, 1.0, 1.0), true, false, &mut f)).expect("failed to write SVG");
-	println!("wrote {svg_path}");
 
+	println!("wrote {example_name}.step / {example_name}.svg");
 	Ok(())
 }
 
@@ -587,13 +579,13 @@ fn build_twisted_ribbon() -> Result<Vec<Solid>, Error> {
 // origin, U-pipe at x=6, ribbon at x=12) and applies its color, so main
 // just concatenates them.
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
+fn main() -> Result<(), Error> {
 	let example_name = std::path::Path::new(file!()).file_stem().unwrap().to_str().unwrap();
 	let all: Vec<Solid> = [build_m2_screw()?, build_u_pipe()?, build_twisted_ribbon()?].concat();
 
-	let mut f = std::fs::File::create(format!("{example_name}.step"))?;
+	let mut f = std::fs::File::create(format!("{example_name}.step")).expect("failed to create STEP file");
 	cadrum::write_step(&all, &mut f)?;
-	let mut f_svg = std::fs::File::create(format!("{example_name}.svg"))?;
+	let mut f_svg = std::fs::File::create(format!("{example_name}.svg")).expect("failed to create SVG file");
 	// Helical threads have dense hidden lines that clutter the SVG; disable them.
 	cadrum::mesh(&all, 0.5)?.write_svg(DVec3::new(1.0, 1.0, -1.0), false, false, &mut f_svg)?;
 	println!("wrote {example_name}.step / {example_name}.svg ({} solids)", all.len());
@@ -659,7 +651,7 @@ fn main() {
 	cadrum::write_step(&objects, &mut f).unwrap();
 	let mut f_svg = std::fs::File::create(format!("{example_name}.svg")).unwrap();
 	cadrum::mesh(&objects, 0.1).and_then(|m| m.write_svg(DVec3::new(0.05, 0.05, 1.0), false, true, &mut f_svg)).unwrap();
-	eprintln!("wrote {0}.step / {0}.svg", example_name);
+	println!("wrote {example_name}.step / {example_name}.svg");
 }
 
 ```

--- a/examples/01_primitives.rs
+++ b/examples/01_primitives.rs
@@ -1,7 +1,6 @@
 //! Primitive solids: box, cylinder, sphere, cone, torus — colored and exported as STEP + SVG.
 
-use cadrum::Solid;
-use glam::DVec3;
+use cadrum::{DVec3, Solid};
 
 fn main() {
     let example_name = std::path::Path::new(file!()).file_stem().unwrap().to_str().unwrap();

--- a/examples/02_write_read.rs
+++ b/examples/02_write_read.rs
@@ -1,7 +1,6 @@
 //! Read and write: chain STEP, BRep text, and BRep binary round-trips with progressive rotation.
 
-use cadrum::{Compound, Solid};
-use glam::DVec3;
+use cadrum::{Compound, DVec3, Solid};
 use std::f64::consts::FRAC_PI_8;
 
 fn main() -> Result<(), cadrum::Error> {

--- a/examples/03_transform.rs
+++ b/examples/03_transform.rs
@@ -1,7 +1,6 @@
 //! Transform operations: translate, rotate, scale, and mirror applied to a cone.
 
-use cadrum::Solid;
-use glam::DVec3;
+use cadrum::{DVec3, Solid};
 use std::f64::consts::PI;
 
 fn main() {

--- a/examples/04_boolean.rs
+++ b/examples/04_boolean.rs
@@ -1,7 +1,6 @@
 //! Boolean operations: union, subtract, and intersect between a box and a cylinder.
 
-use cadrum::{Compound, Solid};
-use glam::DVec3;
+use cadrum::{Compound, DVec3, Solid};
 
 fn main() -> Result<(), cadrum::Error> {
     let example_name = std::path::Path::new(file!()).file_stem().unwrap().to_str().unwrap();

--- a/examples/05_extrude.rs
+++ b/examples/05_extrude.rs
@@ -65,15 +65,12 @@ fn main() -> Result<(), Error> {
 
 	let result = [box_solid, oblique, l_beam, heart];
 
-	let step_path = format!("{example_name}.step");
-	let mut f = std::fs::File::create(&step_path).expect("failed to create STEP file");
+	let mut f = std::fs::File::create(format!("{example_name}.step")).expect("failed to create STEP file");
 	cadrum::write_step(&result, &mut f).expect("failed to write STEP");
-	println!("wrote {step_path}");
 
-	let svg_path = format!("{example_name}.svg");
-	let mut f = std::fs::File::create(&svg_path).expect("failed to create SVG file");
+	let mut f = std::fs::File::create(format!("{example_name}.svg")).expect("failed to create SVG file");
 	cadrum::mesh(&result, 0.5).and_then(|m| m.write_svg(DVec3::new(1.0, 1.0, 1.0), true, false, &mut f)).expect("failed to write SVG");
-	println!("wrote {svg_path}");
 
+	println!("wrote {example_name}.step / {example_name}.svg");
 	Ok(())
 }

--- a/examples/05_extrude.rs
+++ b/examples/05_extrude.rs
@@ -5,8 +5,7 @@
 //! - **L-beam**: L-shaped polygon extruded along Z
 //! - **Heart**: BSpline heart-shaped profile extruded along Z
 
-use cadrum::{BSplineEnd, Edge, Error, Solid};
-use glam::DVec3;
+use cadrum::{BSplineEnd, DVec3, Edge, Error, Solid};
 
 /// Square polygon → box (simplest extrude).
 fn build_box() -> Result<Solid, Error> {

--- a/examples/06_loft.rs
+++ b/examples/06_loft.rs
@@ -47,15 +47,12 @@ fn main() -> Result<(), Error> {
 
 	let result = [frustum, morph, tilted];
 
-	let step_path = format!("{example_name}.step");
-	let mut f = std::fs::File::create(&step_path).expect("failed to create STEP file");
+	let mut f = std::fs::File::create(format!("{example_name}.step")).expect("failed to create STEP file");
 	cadrum::write_step(&result, &mut f).expect("failed to write STEP");
-	println!("wrote {step_path}");
 
-	let svg_path = format!("{example_name}.svg");
-	let mut f = std::fs::File::create(&svg_path).expect("failed to create SVG file");
+	let mut f = std::fs::File::create(format!("{example_name}.svg")).expect("failed to create SVG file");
 	cadrum::mesh(&result, 0.5).and_then(|m| m.write_svg(DVec3::new(1.0, 1.0, 1.0), true, false, &mut f)).expect("failed to write SVG");
-	println!("wrote {svg_path}");
 
+	println!("wrote {example_name}.step / {example_name}.svg");
 	Ok(())
 }

--- a/examples/06_loft.rs
+++ b/examples/06_loft.rs
@@ -4,8 +4,7 @@
 //! - **Morph**: square polygon → circle (cross-section shape transition)
 //! - **Tilted**: three non-parallel circular sections → twisted loft
 
-use cadrum::{Edge, Error, Solid};
-use glam::DVec3;
+use cadrum::{DVec3, Edge, Error, Solid};
 
 /// Two circles → frustum (minimal loft example).
 fn build_frustum() -> Result<Solid, Error> {

--- a/examples/07_sweep.rs
+++ b/examples/07_sweep.rs
@@ -115,13 +115,13 @@ fn build_twisted_ribbon() -> Result<Vec<Solid>, Error> {
 // origin, U-pipe at x=6, ribbon at x=12) and applies its color, so main
 // just concatenates them.
 
-fn main() -> Result<(), Box<dyn std::error::Error>> {
+fn main() -> Result<(), Error> {
 	let example_name = std::path::Path::new(file!()).file_stem().unwrap().to_str().unwrap();
 	let all: Vec<Solid> = [build_m2_screw()?, build_u_pipe()?, build_twisted_ribbon()?].concat();
 
-	let mut f = std::fs::File::create(format!("{example_name}.step"))?;
+	let mut f = std::fs::File::create(format!("{example_name}.step")).expect("failed to create STEP file");
 	cadrum::write_step(&all, &mut f)?;
-	let mut f_svg = std::fs::File::create(format!("{example_name}.svg"))?;
+	let mut f_svg = std::fs::File::create(format!("{example_name}.svg")).expect("failed to create SVG file");
 	// Helical threads have dense hidden lines that clutter the SVG; disable them.
 	cadrum::mesh(&all, 0.5)?.write_svg(DVec3::new(1.0, 1.0, -1.0), false, false, &mut f_svg)?;
 	println!("wrote {example_name}.step / {example_name}.svg ({} solids)", all.len());

--- a/examples/07_sweep.rs
+++ b/examples/07_sweep.rs
@@ -19,8 +19,7 @@
 //!   toward a parallel auxiliary spine. Arbitrary twist control — e.g. a
 //!   helical `aux_spine` on a straight `spine` produces a twisted ribbon.
 
-use cadrum::{Compound, Edge, Error, ProfileOrient, Solid, Wire};
-use glam::DVec3;
+use cadrum::{Compound, DVec3, Edge, Error, ProfileOrient, Solid, Wire};
 
 // ==================== Component 1: M2 ISO screw ====================
 
@@ -90,11 +89,12 @@ fn build_u_pipe() -> Result<Vec<Solid>, Error> {
 
 // ==================== Component 3: Auxiliary-spine twisted ribbon ====================
 
-// 直線 spine を `Auxiliary(&[helix])` で掃引すると、各点で profile の tracked 軸が
-// 対応するヘリックス点を向くように回転される。pitch=h のヘリックスは [0, h] の
-// あいだにちょうど 360° 一周するので、平たい長方形 profile は 1 回捻れた
-// リボンになる — `Fixed` や `Torsion` だと直線 spine では profile は全く
-// 回転しないので、ねじれが見えれば Auxiliary が効いている証拠。
+// Sweeping a straight spine with `Auxiliary(&[helix])` rotates the tracked
+// axis of the profile at each point to face the corresponding helix point.
+// A pitch=h helix makes exactly one 360° turn over [0, h], so a flat
+// rectangular profile becomes a ribbon twisted once. With `Fixed` or
+// `Torsion` the profile wouldn't rotate along a straight spine — visible
+// twist is therefore proof that Auxiliary is in effect.
 fn build_twisted_ribbon() -> Result<Vec<Solid>, Error> {
 	let h = 8.0;
 	let aux_r = 3.0;
@@ -102,7 +102,7 @@ fn build_twisted_ribbon() -> Result<Vec<Solid>, Error> {
 	let spine = Edge::line(DVec3::ZERO, DVec3::Z * h)?;
 	let aux = Edge::helix(aux_r, h, h, DVec3::Z, DVec3::X)?;
 
-	// 平たい長方形 (10:1 アスペクト) — 円や正方形ではねじれが見えない。
+	// Flat rectangle (10:1 aspect) — circles or squares wouldn't reveal any twist.
 	let profile = Edge::polygon(&[DVec3::new(-2.0, -0.2, 0.0), DVec3::new(2.0, -0.2, 0.0), DVec3::new(2.0, 0.2, 0.0), DVec3::new(-2.0, 0.2, 0.0)])?;
 
 	let ribbon = Solid::sweep(&profile, &[spine], ProfileOrient::Auxiliary(&[aux]))?;

--- a/examples/08_bspline.rs
+++ b/examples/08_bspline.rs
@@ -43,5 +43,5 @@ fn main() {
 	cadrum::write_step(&objects, &mut f).unwrap();
 	let mut f_svg = std::fs::File::create(format!("{example_name}.svg")).unwrap();
 	cadrum::mesh(&objects, 0.1).and_then(|m| m.write_svg(DVec3::new(0.05, 0.05, 1.0), false, true, &mut f_svg)).unwrap();
-	eprintln!("wrote {0}.step / {0}.svg", example_name);
+	println!("wrote {example_name}.step / {example_name}.svg");
 }

--- a/examples/08_bspline.rs
+++ b/examples/08_bspline.rs
@@ -1,5 +1,4 @@
-use cadrum::Solid;
-use glam::{DQuat, DVec3};
+use cadrum::{DQuat, DVec3, Solid};
 use std::f64::consts::TAU;
 
 // 2 field-period stellarator-like torus.

--- a/examples/chijin.rs
+++ b/examples/chijin.rs
@@ -57,15 +57,12 @@ fn main() -> Result<(), cadrum::Error> {
 	let example_name = std::path::Path::new(file!()).file_stem().unwrap().to_str().unwrap();
 	let result = [chijin()?];
 
-	let step_path = format!("{example_name}.step");
-	let mut f = std::fs::File::create(&step_path).expect("failed to create STEP file");
+	let mut f = std::fs::File::create(format!("{example_name}.step")).expect("failed to create STEP file");
 	cadrum::write_step(&result, &mut f).expect("failed to write STEP");
-	println!("wrote {step_path}");
 
-	let svg_path = format!("{example_name}.svg");
-	let mut f = std::fs::File::create(&svg_path).expect("failed to create SVG file");
+	let mut f = std::fs::File::create(format!("{example_name}.svg")).expect("failed to create SVG file");
 	cadrum::mesh(&result, 0.5).and_then(|m| m.write_svg(DVec3::new(1.0, 1.0, 1.0), true, false, &mut f)).expect("failed to write SVG");
-	println!("wrote {svg_path}");
 
+	println!("wrote {example_name}.step / {example_name}.svg");
 	Ok(())
 }

--- a/examples/chijin.rs
+++ b/examples/chijin.rs
@@ -1,7 +1,6 @@
 //! Build a chijin (hand drum from Amami Oshima) with colors, boolean ops, and SVG export.
 
-use cadrum::{Color, Compound, Edge, ProfileOrient, Solid};
-use glam::DVec3;
+use cadrum::{Color, Compound, DVec3, Edge, ProfileOrient, Solid};
 use std::f64::consts::PI;
 
 pub fn chijin() -> Result<Solid, cadrum::Error> {

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -34,7 +34,13 @@ pub use occt::solid::Solid;
 pub use common::color::Color;
 pub use common::error::Error;
 pub use common::mesh::{EdgeData, Mesh};
-pub use glam::DVec3;
+// Re-export glam types used in cadrum's public API. Users should reach glam
+// through these re-exports (or the `cadrum::glam` module below) instead of
+// adding a direct `glam` dependency — otherwise a mismatched glam minor
+// version pulls in two incompatible `glam` crates and call sites fail with
+// "expected `cadrum::DVec3`, found `glam::DVec3`".
+pub use glam::{DMat3, DMat4, DQuat, DVec2, DVec3};
+pub use glam;
 
 // ==================== Boolean metadata helpers ====================
 //

